src/app.py

```diff
@@ -81,14 +84,14 @@ def connect(self, **kwargs):
             colSpec = CollectionSpec(self.collection_name, self.scope_name)
             self._bucket.collections().create_collection(colSpec)
 
+            # create index if it doesn't exist
+            # sleep to ensure that the operations are finished before trying to create the index
+            time.sleep(6)
```

Comment:
Do not sleep before letting the user know that you are waiting for something to happen